src/time/format/parser.cr

```diff
@@ -502,6 +502,14 @@ struct Time::Format
     end
 
     def char(char, *alternatives)
+      unless @reader.has_next?
+        if alternatives.empty?
+          raise "Expected #{char.inspect} but the end of the input was reached"
+        else
+          raise "Expected one of #{({char} + alternatives).join(", ", &.inspect)} but reached the input end"
```

Comment:
Optional suggestion that seems a bit more clearer to me, but no strong feelings either way.

```suggestion
          raise "Expected one of #{char.inspect}, #{alternatives.join(", ", &.inspect)} but reached the input end"
```